sycl/include/CL/sycl/buffer.hpp

```diff
@@ -181,8 +181,10 @@ class buffer {
       : Range{0} {
 
     size_t BufSize = 0;
-    PI_CALL(piMemGetInfo)(detail::pi::cast<detail::RT::PiMem>(MemObject),
-                          CL_MEM_SIZE, sizeof(size_t), &BufSize, nullptr);
+    auto Plugin = detail::getSyclObjImpl(SyclContext)->getPlugin();
```

Comment:
Please, do not use auto in such cases
See: https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable.

Reply:
hmmm.. Are you asking me to remove auto keyword since the namespace is different and its an implementations specific class in sycl::buffer class?

Reply:
No, I'm asking you to do this because the type which getPlugin returns is not clear from the context.

Reply:
Could you please apply to the whole patch. Some of the places where I see similar problem:
```
    3 image_impl.hpp : 237, 355
    4 queue_impl.hpp: 91
    5 function_pointer.hpp: 83
    6 device_impl.hpp: 60, 79, 113
    7 kernel_program_cache.cpp: 34, 39
    8 memory_manager.cpp: 28, 62, 105, 126, 145, 196, 231, 289, 314, 420, 476, 493, 508, 524, 539
    9 platform_impl.cpp: 38, 214
   10 program_impl.cpp: 71, 85, 147, 157, 195, 222, 261, 284, 292,
   11 ...

```

Reply:
I have removed the auto keyword from all locations.